eng/packages/http-client-csharp-mgmt/generator/Azure.Generator.Management/src/Utilities/CSharpTypeExtensions.cs

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.TypeSpec.Generator.Primitives;
+using System;
+using System.Linq;
+
+namespace Azure.Generator.Management.Utilities
+{
+    internal static class CSharpTypeExtensions
+    {
+        public static CSharpType WrapAsync(this CSharpType type, bool isAsync)
+        {
+            return isAsync
+                ? new CSharpType(typeof(System.Threading.Tasks.Task<>), type)
+                : type;
+        }
+
+        public static CSharpType WrapResponse(this CSharpType? type, bool isLongRunning)
+        {
+            if (isLongRunning)
+            {
+                return type is not null
+                    ? new CSharpType(typeof(Azure.ResourceManager.ArmOperation<>), type)
+                    : typeof(Azure.ResourceManager.ArmOperation);
+            }
+            else
+            {
+                return type is not null
+                    ? new CSharpType(typeof(Azure.Response<>), type)
+                    : typeof(Azure.Response);
+            }
+        }
+
+        public static CSharpType UnWrapAsync(this CSharpType type)
+        {
+            if (type.IsGenericType && type.Name == "Task")
```

Comment:
I think for types, comparing with names would always be problematic.
We could do this instead:
```
type.IsFramework && type.FrameworkType.GetGenericTypeDefinition() == typeof(Task<>)
```
to 100% sure identify the Task type, instead of catching some model with the name of `Task` for example